src/mito2/src/cache.rs

```diff
@@ -69,15 +70,30 @@ impl CacheManager {
     }
 
     /// Gets cached [ParquetMetaData].
-    pub fn get_parquet_meta_data(
+    pub async fn get_parquet_meta_data(
         &self,
         region_id: RegionId,
         file_id: FileId,
     ) -> Option<Arc<ParquetMetaData>> {
-        self.sst_meta_cache.as_ref().and_then(|sst_meta_cache| {
+        // Try to get metadata from sst meta cache
+        let metadata = self.sst_meta_cache.as_ref().and_then(|sst_meta_cache| {
             let value = sst_meta_cache.get(&SstMetaKey(region_id, file_id));
             update_hit_miss(value, SST_META_TYPE)
-        })
+        });
+
+        if metadata.is_some() {
+            return metadata;
+        }
+
+        // Try to get metadata from write cache
+        let key = IndexKey::new(region_id, file_id, FileType::Parquet);
+        if let Some(write_cache) = &self.write_cache {
+            if let Some(metadata) = write_cache.file_cache().get_parquet_meta_data(key).await {
+                return Some(Arc::new(metadata));
+            }
```

Comment:
Will anyone put the metadata obtained from the `write_cache` back to the `sst_meta_cache`?

Reply:
good catch!